pkg/controller/machine-set-boot-image/machine_set_boot_image_controller.go

```diff
@@ -312,20 +315,74 @@ func (ctrl *Controller) updateConfigMap(oldCM, newCM interface{}) {
 		return
 	}
 
-	// Only take action if the configmap.ResourceVersion has bumped indicating a change in the resource has taken place.
-	// This callback seems to take place fairly frequently when there has been no changes in the resource itself
-	// and this check helps avoid unnecessary full loop syncs.
+	// Only take action if the there is an actual change in the configMap Object
 	if oldConfigMap.ResourceVersion == newConfigMap.ResourceVersion {
 		return
 	}
 
-	klog.Infof("configMap %s updated, reconciling all machine sets", oldConfigMap.Name)
+	klog.Infof("configMap %s updated, reconciling enrolled machine resources", oldConfigMap.Name)
 
 	// Update all machine sets since the "golden" configmap has been updated
-	ctrl.enqueueAllMachineSets()
+	err := ctrl.enqueueMAPIMachineSets()
+	if err != nil {
+		klog.Errorf("Error enqueuing MAPI machine sets: %v", err)
+	}
 }
 
-func (ctrl *Controller) enqueue(ms *machinev1beta1.MachineSet) {
+func (ctrl *Controller) addMachineConfiguration(obj interface{}) {
+	// No-op if feature is disabled
+	if !ctrl.featureEnabled {
+		return
+	}
+
+	machineConfiguration := obj.(*opv1.MachineConfiguration)
+
+	// Take no action if this isn't the "cluster" level MachineConfiguration object
+	if machineConfiguration.Name != ctrlcommon.MCOOperatorKnobsObjectName {
+		klog.V(4).Infof("MachineConfiguration %s updated, but does not match %s, skipping bootimage sync", machineConfiguration.Name, ctrlcommon.MCOOperatorKnobsObjectName)
+		return
+	}
+
+	klog.Infof("Bootimages management configuration has been updated, reconciling enrolled machine resources")
```

Comment:
General syncs should try to avoid logging the same message on every loop (this gated by tech preview so it's fine for now)

Reply:
I can bump up the levels on these (:

Reply:
Actually on second thought - I think we should hold off until GA. It is useful to see why the sync loop is happening. I've added a deepequal check on the boot images knob so hopefully we shouldn't see it very often